scripts/cnode-helper-scripts/env

```diff
@@ -297,4 +282,21 @@ STANDOUT='\e[7m'
 BOLD='\e[1m'
 NC='\e[0m'
 
+getEraIdentifier() {
+    if ${CCLI} query protocol-parameters --byron-era ${PROTOCOL_IDENTIFIER} ${NETWORK_IDENTIFIER} &>/dev/null; then ERA_IDENTIFIER="--byron-era" && ERA_WITNESS="TxWitness ByronEra"
+  elif ${CCLI} query protocol-parameters --shelley-era ${PROTOCOL_IDENTIFIER} ${NETWORK_IDENTIFIER} &>/dev/null; then ERA_IDENTIFIER="--shelley-era" && ERA_WITNESS="TxWitness ShelleyEra"
+  elif ${CCLI} query protocol-parameters --allegra-era ${PROTOCOL_IDENTIFIER} ${NETWORK_IDENTIFIER} &>/dev/null; then ERA_IDENTIFIER="--allegra-era" && ERA_WITNESS="TxWitness AllegraEra"
+  elif ${CCLI} query protocol-parameters --mary-era ${PROTOCOL_IDENTIFIER} ${NETWORK_IDENTIFIER} &>/dev/null; then ERA_IDENTIFIER="--mary-era" && ERA_WITNESS="TxWitness MaryEra"
+  else 
+    ERA_IDENTIFIER=""; ERA_WITNESS=""
+    [[ ${OFFLINE_MODE} = "N" ]] && return 1
+  fi
+  return 0
+}
+# special exit code for scripts that want to source env and doesn't care about ERA_IDENTIFIER or ERA_WITNESS
+if ! getEraIdentifier && [[ ${OFFLINE_MODE} = "N" ]]; then echo "Failed to query protocol-parameters from node, not yet fully started?" && return 2; fi
```

Comment:
Nothing big, but just noticed we have OFFLINE_MODE check already on line 292, and essentially getEraIdentifier will return 1 only if  ${OFFLINE_MODE} = "N".
Possibly below?
```suggestion
[[ ! getEraIdentifier ]] && echo "Failed to query protocol-parameters from node, not yet fully started?" && return 2
```

Reply:
Correct, as `[[ ${OFFLINE_MODE} = "N" ]]` is checked in `getEraIdentifier()` it will always return 0 in offline mode so no need to check this again. Though function call can not be used inside `[[ ]]` so I suggest it will be changed to this instead:
`if ! getEraIdentifier; then echo "Failed to query protocol-parameters from node, not yet fully started?" && return 2; fi`

Reply:
agreed :)